core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/UserResource.scala

```diff
@@ -97,7 +185,11 @@ class UserResource {
   @GET
   @Path("/logout")
   def logOut(@Session session: HttpSession): Response = {
-    setUserSession(session, null)
+    if (isGoogleUser(session)) {
+      setGoogleUserSession(session, null)
+    } else {
+      setUserSession(session, null)
+    }
```

Comment:
just curious, could there be a case where the user is logged in both as a normal user and a google user?
I actually suggest you set both to null in either case.
```suggestion
    setGoogleUserSession(session, null)
    setUserSession(session, null)
```